In [1]:
__author__ = "Sreejith Sreekumar"
__email__ = "sreekumar.s@husky.neu.edu"
__version__ = "0.0.1"

In [2]:
%load_ext autoreload
%autoreload 2
%matplotlib inline
%debug

import warnings
warnings.filterwarnings('ignore')

ERROR:root:No traceback has been produced, nothing to debug.


In [3]:
import pandas as pd
import numpy as np
from sklearn.preprocessing import StandardScaler, OneHotEncoder, LabelEncoder
from sklearn.neighbors import KNeighborsClassifier
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import roc_auc_score, balanced_accuracy_score, f1_score
from sklearn.metrics import make_scorer
import seaborn as sns
sns.set()

from sklearn.model_selection import train_test_split

import matplotlib.pyplot as plt
import plotly.plotly as py
import plotly.graph_objs as go

In [4]:
data_path = "../data/winemag-data_first150k.csv"
data = pd.read_csv(data_path)

In [5]:
data.head(5)

Unnamed: 0 country                                        description  \
0           0      US  This tremendous 100% varietal wine hails from ...   
1           1   Spain  Ripe aromas of fig, blackberry and cassis are ...   
2           2      US  Mac Watson honors the memory of a wine once ma...   
3           3      US  This spent 20 months in 30% new French oak, an...   
4           4  France  This is the top wine from La Bégude, named aft...   

                            designation  points  price        province  \
0                     Martha's Vineyard      96  235.0      California   
1  Carodorum Selección Especial Reserva      96  110.0  Northern Spain   
2         Special Selected Late Harvest      96   90.0      California   
3                               Reserve      96   65.0          Oregon   
4                            La Brûlade      95   66.0        Provence   

            region_1           region_2             variety  \
0        Napa Valley               Napa  Cabernet Sauvignon   
1               Toro                NaN       Tinta de Toro   
2     Knights Valley             Sonoma     Sauvignon Blanc   
3  Willamette Valley  Willamette Valley          Pinot Noir   
4             Bandol                NaN  Provence red blend   

                    winery  
0                    Heitz  
1  Bodega Carmen Rodríguez  
2                 Macauley  
3                    Ponzi  
4     Domaine de la Bégude

__Visualization1: What is the average wine price per country?__

In [6]:
grouped_avg_price = data.groupby("country")["price"].mean().reset_index(name="average_price")
grouped_avg_price = grouped_avg_price.sort_values(by="average_price", ascending=False)
grouped_avg_price = grouped_avg_price.dropna()


_data = [go.Line(
            x=grouped_avg_price["country"],
            y=grouped_avg_price["average_price"]
    )]



layout = go.Layout(
    title='Average price of wine country-wise',
    
    xaxis=dict(
        title='Country',
        titlefont=dict(
            family='Lato',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=dict(
        title='Price of wine',
        titlefont=dict(
            family='Lato',
            size=18,
            color='#7f7f7f'
        )
    )
)

fig = go.Figure(data=_data, layout=layout)
py.iplot(fig, filename='price')

/usr/local/lib/python3.6/dist-packages/plotly/graph_objs/_deprecations.py:372: DeprecationWarning:

plotly.graph_objs.Line is deprecated.
Please replace it with one of the following more specific types
  - plotly.graph_objs.scatter.Line
  - plotly.graph_objs.layout.shape.Line
  - etc.




__Which country produces the most number of `unique` wines__

(What % of the unique wines are made by a country)

In [7]:
unique_wines = data.groupby(["variety","country"]).size().reset_index(name="count")
unique_wines = unique_wines[["variety","country"]]

In [8]:
wines_count = unique_wines.groupby("country").size().reset_index(name="count")
total_number_of_wines = sum(wines_count["count"])

In [9]:
# Number of unique wines in the dataset
len(data["variety"].unique())

632

__There are 632 unique wines in the dataset__

What % of all the unique wines are made by a country?

In [10]:
wines_count = wines_count.sort_values(by="count",ascending=False)

_data = [go.Bar(
            x=wines_count["country"],
            y=wines_count["count"]
    )]


layout = go.Layout(
    title='Number of unique wines available country-wise',
    
    xaxis=dict(
        title='Country',
        titlefont=dict(
            family='Lato',
            size=18,
            color='#7f7f7f'
        )
    ),
    yaxis=dict(
        title='Unique Wines Made',
        titlefont=dict(
            family='Lato',
            size=18,
            color='#7f7f7f'
        )
    )
)

fig = go.Figure(data=_data, layout=layout)
py.iplot(fig, filename='Number of unique wines')